packages/object-extend/index.js

```diff
@@ -28,12 +29,18 @@ module.exports = extend;
   extend(3, {a: 4, b: 5}); // throws
 */
 
-function extend(/* [deep], obj1, obj2, [objn] */) {
+function extend(/* [deep], obj1, obj2, [objn], new WeakMap() */) {
   var args = [].slice.call(arguments);
   var deep = false;
   if (typeof args[0] == 'boolean') {
     deep = args.shift();
   }
+  var weakMap;
```

Comment:
Unfortunately `WeakMap` breaks our support matrix. I should probably revisit the support matrix shortly, but in the meantime I think we might have to leave circular deps out.